api/provider/aws/test_aws/service_create_test.go

```diff
@@ -187,26 +233,220 @@ func TestServiceCreate_defaults(t *testing.T) {
+		{
+			EntityID:   "svc_id",
+			EntityType: "service",
+			Key:        "name",
+			Value:      "svc_name",
+		},
+		{
+			EntityID:   "svc_id",
+			EntityType: "service",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+	}
+
+	for _, tag := range expectedTags {
+		assert.Contains(t, tagStore.Tags(), tag)
+	}
+}
+
+func TestServiceCreate_stateful(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
 	defer provider.SetEntityIDGenerator("svc_id")()
 
+	tags := models.Tags{
+		{
+			EntityID:   "dpl_id",
+			EntityType: "deploy",
+			Key:        "arn",
+			Value:      "dpl_arn",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	// define expected ecs.DescribeTaskDefinition
+	taskDefinitionInput := &ecs.DescribeTaskDefinitionInput{}
+	taskDefinitionInput.SetTaskDefinition("dpl_arn")
+
+	portMapping := &ecs.PortMapping{}
+	portMapping.SetContainerPort(int64(80))
+
+	portMappings := []*ecs.PortMapping{
+		portMapping,
+	}
+
+	containerDefinition := &ecs.ContainerDefinition{}
+	containerDefinition.SetName("ctn_name")
+	containerDefinition.SetPortMappings(portMappings)
+
+	containerDefinitions := []*ecs.ContainerDefinition{
+		containerDefinition,
+	}
+
+	taskDefinition := &ecs.TaskDefinition{
+		ContainerDefinitions: containerDefinitions,
+	}
+
+	taskDefinitionOutput := &ecs.DescribeTaskDefinitionOutput{}
+	taskDefinitionOutput.SetTaskDefinition(taskDefinition)
+
+	mockAWS.ECS.EXPECT().
+		DescribeTaskDefinition(taskDefinitionInput).
+		Return(taskDefinitionOutput, nil)
+
+	// define expected CreateService
 	createServiceInput := &ecs.CreateServiceInput{}
 	createServiceInput.SetCluster("l0-test-env_id")
 	createServiceInput.SetDesiredCount(1)
+	createServiceInput.SetLaunchType(ecs.LaunchTypeEc2)
 	createServiceInput.SetServiceName("l0-test-svc_id")
 	createServiceInput.SetTaskDefinition("dpl_arn")
 
 	mockAWS.ECS.EXPECT().
 		CreateService(createServiceInput).
 		Return(&ecs.CreateServiceOutput{}, nil)
 
+	// define request
 	req := models.CreateServiceRequest{
 		DeployID:      "dpl_id",
 		EnvironmentID: "env_id",
 		ServiceName:   "svc_name",
+		ServiceType:   models.DeployCompatibilityStateful,
 	}
 
 	target := provider.NewServiceProvider(mockAWS.Client(), tagStore, mockConfig)
-	if _, err := target.Create(req); err != nil {
+	result, err := target.Create(req)
+	if err != nil {
 		t.Fatal(err)
 	}
+
+	assert.Equal(t, result, "svc_id")
+
+	expectedTags := models.Tags{
+		{
+			EntityID:   "svc_id",
+			EntityType: "service",
+			Key:        "name",
+			Value:      "svc_name",
+		},
+		{
+			EntityID:   "svc_id",
+			EntityType: "service",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+	}
+
+	for _, tag := range expectedTags {
```

Comment:
Same as the comment above. It would be good if we start checking the count for the tagstore values to match the expected value of 2 in this case.

Reply:
✅